pkg/p2p/kadcast/writer.go

```diff
@@ -115,51 +116,79 @@ func (w *Writer) WriteToPoint(m message.Message) error {
 
 	// Constuct gossip frame.
 	buf := m.Payload().(message.SafeBuffer)
-	if err := w.gossip.Process(&buf.Buffer); err != nil {
+	if err = w.gossip.Process(&buf.Buffer); err != nil {
 		log.WithError(err).Error("reading gossip frame failed")
 		return err
 	}
 
+	// Marshal message data
+	var packet []byte
+
+	packet, err = w.marshalBroadcastPacket(height, buf.Bytes())
+	if err != nil {
+		log.WithError(err).Warn("marshal broadcast packet failed")
+	}
+
 	// Send message to a single destination using height = 0.
-	w.sendToDelegates(delegates, height, buf.Bytes())
+	w.sendToDelegates(delegates, height, packet)
 	return nil
 }
 
 // BroadcastPacket sends a `CHUNKS` message across the network
 // following the Kadcast broadcasting rules with the specified height.
-func (w *Writer) broadcastPacket(height byte, payload []byte) {
-	router := w.router
-	myPeer := router.LpeerInfo
-
-	if height > byte(len(router.tree.buckets)) || height == 0 {
+func (w *Writer) broadcastPacket(maxHeight byte, payload []byte) {
+	if maxHeight > byte(len(w.router.tree.buckets)) || maxHeight == 0 {
 		return
```

Comment:
Would it be useful to return an error here, to let the caller know that we aren't broadcasting a packet?

Reply:
That actually makes me reconsider error reporting in general. Addressed it with latest commit.